### PR TITLE
Add interface for notification handler for dimissed events

### PR DESCRIPTION
**Pull request by rullzer**
For https://github.com/nextcloud/notifications/issues/477

### CODE REVIEW

**Comment by nickvergessen** [comment]
:+1: but the name feels weird. It should reflect that it belongs to INotifier?

**Comment by rullzer** [comment]
> +1 but the name feels weird. It should reflect that it belongs to INotifier?

other suggestions?

**Comment by nickvergessen** [comment]
In User/group backends we call them I*Backend.
So `IDismissNotifier` would match the pattern, or something like `IOnDismissNotifier`

**Comment by rullzer** [comment]
Lets go for `IDismissNotifier` :)

**Comment by ChristophWurst** [comment]
`IDismissableNotifier`?

**Comment by rullzer** [comment]
I like that even better

File: lib/public/Notification/IHandleDismissedNotification.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+namespace OCP\Notification;
+
+/**
+ * Interface INotifier classes should implement if they want to process notifications
+ * that are dismissed by the user.
+ *
+ * This can be useful if dismissing the notification will leave it in an incomplete
+ * state. The handler can chose to for example do some default action.
+ */
+interface IHandleDismissedNotification {
```
Comment (ChristophWurst) [inline lib/public/Notification/IHandleDismissedNotification.php]:
```suggestion
interface IHandleDismissedNotification extends INotifier {
```

Then you can assume an object is both if you do

```php
if ($notifier instanceof IHandleDismissedNotification) {
 ...
}
```

Comment (ChristophWurst) [reply lib/public/Notification/IHandleDismissedNotification.php]: (just like https://github.com/nextcloud/server/blob/63a2725f3d38fe8e5e767cd62f65330776435811/lib/public/Authentication/TwoFactorAuth/IActivatableAtLogin.php#L34 and friends)

Comment (nickvergessen) [reply lib/public/Notification/IHandleDismissedNotification.php]: But we can't have more Interfaces in the future, if they are not linear?

Comment (rullzer) [reply lib/public/Notification/IHandleDismissedNotification.php]: @nickvergessen well you can if they all extend the base notifier

Comment (nickvergessen) [reply lib/public/Notification/IHandleDismissedNotification.php]: Right, brain fail

**Comment by rullzer** [comment]
See also https://github.com/nextcloud/notifications/pull/508

**Review by ChristophWurst — APPROVED** [review]
Looks good!

**Comment by rullzer** [comment]
rebased and squased
@nickvergessen please give it your final look ;)

File: lib/public/Notification/IManager.php [hunk 2]
```diff
@@ -85,4 +85,10 @@ public function setPreparingPushNotification(bool $preparingPushNotification): v
 	 * @since 14.0.0
 	 */
 	public function isPreparingPushNotification(): bool;
+
+	/**
+	 * @return void
```
Comment (nickvergessen) [inline lib/public/Notification/IManager.php]: Normally we dont document void, but well :P

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**